### PR TITLE
Decouple neon-lib and neon-cli in separate crates

### PR DESCRIPTION
**Pull request by Denommus**

### CODE REVIEW

File: evm_loader/lib/src/config.rs
Comment (dk-neon) [inline evm_loader/lib/src/config.rs:1]: I think, we should leave functions which work with `ArgMatches`, in CLI.

File: evm_loader/lib/src/context.rs
Comment (dk-neon) [inline evm_loader/lib/src/context.rs:1]: Same as before. I think, we should leave functions which work with ArgMatches, in CLI.

File: evm_loader/lib/src/program_options.rs
Comment (dk-neon) [inline evm_loader/lib/src/program_options.rs:1]: Same as before. I think, we should leave functions which work with ArgMatches, in CLI.

File: evm_loader/cli/src/api_server/handlers/mod.rs [hunk 1]
```diff
@@ -74,19 +73,6 @@ pub(crate) fn parse_emulation_params(
     (token, chain, max_steps, accounts, solana_accounts)
 }
 
-impl IntoResponse for NeonCliError {
```
Comment (dk-neon) [inline evm_loader/cli/src/api_server/handlers/mod.rs:77]: I think, we should make additional binary, for the API server, which will use neon_lib as dependency. So we will have both CLI and API server executables, each of them will work its own way (CLI for command line and API server for handling REST requests).

**Review by dk-neon — CHANGES_REQUESTED** [review]

File: evm_loader/cli/src/api_server/handlers/mod.rs (hunk 1 above)
Comment (Denommus) [reply evm_loader/cli/src/api_server/handlers/mod.rs:77]: The API server is a different binary already, check `cli/Cargo.toml`

Comment (dk-neon) [reply evm_loader/cli/src/api_server/handlers/mod.rs:77]: Yes, you are right, I had missed the thing that it is different binaries. But at this moment both of them are placed in the same crate. How about splitting them to separate crates and remove excessive dependencies?

Comment (Denommus) [reply evm_loader/cli/src/api_server/handlers/mod.rs:77]: Ok, understood.

File: evm_loader/lib/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+pub mod event_listener;
+pub mod rpc;
+pub mod syscall_stubs;
+pub mod types;
+
+pub use config::Config;
+pub use context::Context;
+pub use errors::NeonCliError;
+
+pub type NeonCliResult = Result<serde_json::Value, NeonCliError>;
```
Comment (dk-neon) [inline evm_loader/lib/src/lib.rs:15]: We should get rid of NeonCliResult in the library. Let it return scalars/structures with solana/evm library errors, and CLI will wrap them to JSON and return NeonCliResult only in CLI functions.

**Review by dk-neon — CHANGES_REQUESTED** [review]

**Review by dk-neon — CHANGES_REQUESTED** [review]

**Pull request merged**